packages/design-system/src/components/TextField/useLabelMask.tsx

```diff
@@ -5,38 +5,102 @@ import { TextInputProps } from './TextInput';
 
 export type MaskFunction = (rawInput: string, valueOnly?: boolean) => string;
 
+/**
+ * Regular expressions for matching commonly used masks, ensure capture groups are
+ * set for each set of items that are separated by a delimiter, delimiters are
+ * optional and so should be shown as well with the `?` optional flag.
+ */
 export const RE_DATE = /^(\d{1,2})[\D]?(\d{1,2})?[\D]?(\d{1,4})?/;
+export const RE_PHONE = /^\(?(\d{1,3})?\)?[\s.-]?(\d{1,3})?[\s.-]?(\d{1,4})?/;
+export const RE_SSN = /([*\d]{1,3})[\s.-]?([*\d]{1,2})?[\s.-]?([\d{1,4}]+)?/;
+export const RE_ZIP = /(\d{1,5})/;
 
 /**
- * Takes the string value from an input and returns a string
- * with appropriate date format masking applied concatenated
- * with the date hint text, MM/DD/YYYY.
- *
- * `valueOnly` defaults to false. If true, returns formatted
- * string without additional hint text.
+ * This function returns a mask function which returns either the formatted match only, or
+ * the formatted match + the hint substring which the regular expressions above match against
+ * when determining what content has been filled in.
  */
-export const DATE_MASK: MaskFunction = (rawInput = '', valueOnly = false) => {
-  const match = RE_DATE.exec(rawInput);
-  let formattedDate = '';
-  if (match) {
-    const [month, day, year] = match.slice(1);
-    formattedDate = [
-      // We treat all non-numeric characters as a delimiter. If they're using a
-      // delimiter after a month or day, we interpret that as the user supplying
-      // a single digit for month or day, which we will automatically pad for them.
-      month && month.padStart(2, '0'),
-      day && day.padStart(2, '0'),
-      year,
-    ]
-      .filter((s) => s)
-      .join('/');
-  }
+const makeMask = (regex: RegExp, hint: string, formatter: (stringMatch: string[]) => string) => {
+  return (rawInput = '', valueOnly = false) => {
+    const match = regex.exec(rawInput);
+    let formattedMatch = '';
+    if (match) {
+      formattedMatch = formatter(match);
+    }
+    if (valueOnly) {
+      return formattedMatch;
+    }
+    const hintSub = hint.substring(formattedMatch.length);
+    return formattedMatch + hintSub;
+  };
+};
+
+/**
+ * The date mask automatically pads months and days that are one digit
+ */
+export const DATE_MASK: MaskFunction = makeMask(RE_DATE, 'MM/DD/YYYY', (match) => {
+  const [month, day, year] = match.slice(1);
+  const formattedDate = [
+    // We treat all non-numeric characters as a delimiter. If they're using a
+    // delimiter after a month or day, we interpret that as the user supplying
+    // a single digit for month or day, which we will automatically pad for them.
+    month && month.padStart(2, '0'),
+    day && day.padStart(2, '0'),
+    year,
+  ]
+    .filter((s) => s)
+    .join('/');
+
+  return formattedDate;
+});
+
+/**
+ * Formatting for US phone numbers
+ */
+export const PHONE_MASK: MaskFunction = makeMask(RE_PHONE, 'XXX-XXX-XXXX', (match) => {
```

Comment:
I think it might be a bit of a pain to get this one to work with the mask being like that, since it uses a different pattern than the others. Personally I think `$#.##` or `$X.XX` both look a little odd compared to `$0.00`

Reply:
I agree `#` makes sense for the digits

Reply:
I feel like it would be better to show nothing since users would understand that it's looking for numbers. The problem is 1) consistency and 2) avoiding content shift if we show and hide a mask.

Reply:
One thing the old one did, too, was only insert the decimals if they were present in the raw input. It's nice to be able to use the Intl module, but some complexity might be necessary here for more nuanced behaviors.

Reply:
This works as is, though, so maybe we should continue to refine this one in a follow-up pull request.

Reply:
It seems like we need a discussion as a team to determine the requirements for this one since it's a modified pattern. Since we won't really be meeting again all together till after Thanksgiving I would agree maybe a followup to change anything that comes up after that consensus would be a good idea.